src/anyp/PortCfg.h

```diff
@@ -72,16 +80,72 @@ class PortCfg : public CodeContext
 
 private:
     explicit PortCfg(const PortCfg &other); // for ipV4clone() needs only!
+    char *name_; ///< internal name for the port
+    char *spec_; ///< the port specification (port or addr:port)
+};
+
+/// Iterates over a PortCfg list and sets the corresponding CodeContext before each iteration.
+class PortIterator
+{
+public:
+    // some of the standard iterator traits
+    using iterator_category = std::forward_iterator_tag;
+    using value_type = PortCfgPointer;
+    using pointer = value_type *;
+    using reference = value_type &;
+
+    /// \param first the PortCfg this iterator points to
+    PortIterator(const PortCfgPointer &first): position_(first) { setContext(); }
+
+    /// Special constructor for end() iterator.
+    /// This iterator does not point to an object and should never be dereferenced.
+    PortIterator(): position_(nullptr) {}
+
+    reference operator *() { return position_; }
+    pointer operator ->() { return &position_; }
```

Comment:
It may seem a bit silly in the current code, but if we start resetting context when the iterator is dereferenced, then we should definitely avoid this (currently minor) code duplication:

```suggestion
    pointer operator ->() { return &(operator*()); }
```

And even in the current code, it would be nice to Assure() position existence before returning the port object at that position, which would increase code duplication.